src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @typedef {Object} VersionNumber
+ * @property {string} major
+ * @property {string} minor
+ * @property {string} patch
+ */
+
+/**
+ * parses a versionNumber string
+ * @returns {VersionNumber}
+ */
+function parseVersionNumber() {
```

Comment:
Hmm that looks subtly to break in case that version display changes.

Can't we just ignore the version number and fallback/break, in case it is no Mastodon instance? Or just check whether it is a Mastodon instance instead of a specific version?

Also note an alternative would be to [fetch the instance version via the official Mastodon API endpoint](https://docs.joinmastodon.org/methods/instance/#fetch-instance).
I would not really prefer that though either, as that would imply we send a new web request to each website/domain you visit once you visit that, which is:
* both a network performance bottleneck (imagine mobile networks) _and_
* potentially privacy risk (though well you visit the site already, but you'd spam "innocent" web servers with requests and expose the [add-on version](https://github.com/rugk/mastodon-simplified-federation/blob/b7259e0baaf87f49ed605116426ec0dd489fb5a5/src/common/modules/NetworkTools.js#L41)

Reply:
Also what happens if instances modify their theme or whatever, e.g. hide the footer or version number (and if they believe that increases their security e.g.).